### PR TITLE
Switch source writer entirely over to using interpolations.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @333fred @jcouv

**Comment by CyrusNajmabadi** [comment]
@jcouv Thanks.  As this makes no changes to actually any emitted source, do you think we can get this merged in soon?

**Comment by CyrusNajmabadi** [comment]
@333fred want to take a look?  fairly minimal change.

File: src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs [hunk 1]
```diff
@@ -472,31 +455,26 @@ private void WriteGreenVisitor(bool withResult)
...
                 var type =
-                    field.Type == "SyntaxNodeOrTokenList" ? "Microsoft.CodeAnalysis.Syntax.InternalSyntax.SyntaxList<CSharpSyntaxNode>" :
-                    field.Type == "SyntaxTokenList" ? "Microsoft.CodeAnalysis.Syntax.InternalSyntax.SyntaxList<SyntaxToken>" :
-                    IsNodeList(field.Type) ? "Microsoft.CodeAnalysis.Syntax.InternalSyntax." + field.Type :
-                    IsSeparatedNodeList(field.Type) ? "Microsoft.CodeAnalysis.Syntax.InternalSyntax." + field.Type :
-                    field.Type;
-
-                Write("{0} {1}", type, CamelCase(field.Name));
-            }
+                    f.Type == "SyntaxNodeOrTokenList" ? "Microsoft.CodeAnalysis.Syntax.InternalSyntax.SyntaxList<CSharpSyntaxNode>" :
```
Comment (RikkiGibson) [inline src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs]: I feel like I'm getting deja vu, did we just change a complex if-else chain similar to this one to a switch expression? Not that it matters that much here, the goal seems to be just mechanical translation right now.

File: src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs [hunk 2]
```diff
@@ -1235,27 +1159,18 @@ private void WriteRedVisitor(bool genericResult)
                 if (nWritten > 0)
                     WriteLine();
                 nWritten++;
-                WriteComment(string.Format("<summary>Called when the visitor visits a {0} node.</summary>", node.Name));
-                WriteLine(
```
Comment (RikkiGibson) [inline src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs]: yeah this mix of concatenation and format arguments is good to move away from.

File: src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs]: yeah, ther'es a lot of opportunities for cleanup/consistency here.  it's never going to happen in one PR, but rather over a set of many steps :)

File: src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs]: :)

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Pull request merged**